### PR TITLE
Md/feat/bond test

### PR DESCRIPTION
**Pull request by DedicatedDev**
- Update Mock contract to test with custom netyield

### CODE REVIEW

File: contracts/strategy/instadapp/instadappLite.sol [hunk 1]
```diff
@@ -718,12 +718,19 @@ abstract contract ERC20 is Context, IERC20, IERC20Metadata, IERC20Errors {
...
 
+
+
 contract StinstaToken is ERC20, Ownable {
     // Event emitted when Ether is deposited and Stinsta tokens are minted
     event Deposit(address indexed depositor, uint256 amount);
     event WithdrawStinstaTokens(address indexed to, uint256 amount);
     address public mstEth;
     mapping(address => uint256) public deposits;
+    mapping(address => uint256) public currentRewards;
```
Comment (Beardev118) [inline contracts/strategy/instadapp/instadappLite.sol:729]: Please rm this part

File: hardhat.config.ts [hunk 2]
```diff
@@ -12,10 +12,10 @@ module.exports = {
         hardhat: {
             allowUnlimitedContractSize: true,
         },
-        mumbai: {
-            url: process.env.RPC,
-            accounts: [`0x${process.env.PVTKEY}`],
-        },
+        // mumbai: {
```
Comment (Beardev118) [inline hardhat.config.ts:15]: Please open this part

File: contracts/EnderTreasury.sol [hunk 3]
```diff
@@ -14,6 +14,7 @@ import "./interfaces/IEndToken.sol";
 import "./interfaces/IInstadappLite.sol";
 import "./interfaces/ILybraFinance.sol";
 import "./interfaces/IEnderBond.sol";
+import "hardhat/console.sol";
```
Comment (Beardev118) [inline contracts/EnderTreasury.sol:17]: rm

File: contracts/EnderTreasury.sol [hunk 4]
```diff
@@ -248,6 +249,7 @@ contract EnderTreasury is Initializable, OwnableUpgradeable, EnderELStrategy {
             IERC20(_asset).approve(_strategy, _depositAmt);
             IInstadappLite(instadapp).deposit(_depositAmt); // note for testing we changed the function sig.
             instaDappDepositValuations += _depositAmt;
+            console.log("deposit:",_depositAmt);
```
Comment (Beardev118) [inline contracts/EnderTreasury.sol:252]: rm

File: contracts/EnderTreasury.sol [hunk 5]
```diff
@@ -365,10 +367,11 @@ contract EnderTreasury is Initializable, OwnableUpgradeable, EnderELStrategy {
 
     function calculateDepositReturn(address _stEthAddress) public view returns (uint256 depositReturn) {
         uint256 totalReturn = calculateTotalReturn(_stEthAddress);
+        console.log("totalReturn:", totalReturn);
```
Comment (Beardev118) [inline contracts/EnderTreasury.sol:370]: rm

File: contracts/EnderTreasury.sol (hunk 3 above)
Comment (DedicatedDev) [reply contracts/EnderTreasury.sol:17]: I guess we can remove this when we've done. 
There are still many things to debug

File: contracts/strategy/instadapp/instadappLite.sol [hunk 6]
```diff
@@ -718,12 +718,18 @@ abstract contract ERC20 is Context, IERC20, IERC20Metadata, IERC20Errors {
 
 // Importing necessary ERC-20 interfaces
 
+
```
Comment (Beardev118) [inline contracts/strategy/instadapp/instadappLite.sol:721]: rm

File: contracts/strategy/instadapp/instadappLite.sol [hunk 7]
```diff
@@ -718,12 +718,18 @@ abstract contract ERC20 is Context, IERC20, IERC20Metadata, IERC20Errors {
...
+
 contract StinstaToken is ERC20, Ownable {
     // Event emitted when Ether is deposited and Stinsta tokens are minted
     event Deposit(address indexed depositor, uint256 amount);
     event WithdrawStinstaTokens(address indexed to, uint256 amount);
     address public mstEth;
     mapping(address => uint256) public deposits;
+    mapping (address => uint256) public lastupdates;
```
Comment (Beardev118) [inline contracts/strategy/instadapp/instadappLite.sol:729]: rm

File: contracts/strategy/instadapp/instadappLite.sol [hunk 8]
```diff
@@ -753,7 +759,7 @@ contract StinstaToken is ERC20, Ownable {
         _mint(msg.sender, stinstaAmount);
 
         // // Update the mapping to track the deposited and minted amounts
-        // deposits[msg.sender] += mstEthAmount;
+        deposits[msg.sender] += mstEthAmount;
```
Comment (Beardev118) [inline contracts/strategy/instadapp/instadappLite.sol:762]: Please disable this line

**Review by Beardev118 — APPROVED** [review]

**Pull request merged**